visions/abi-stability.md

```diff
@@ -191,6 +198,42 @@ package builds take this approach, but have to restrict themselves as
 a result to a small subset of libraries that are known to be ABI
 stable.
 
+## Declaring ABI Stability
+
+Supported platforms for Swift must have a Platform Owner.  Because a
+declaration of ABI stability for a platform has implications for
+compiler and runtime engineers who do not develop on that platform, we
+intend to set a high bar for declaring ABI stability.  As such, to
+become officially ABI stable:
+
+1. The Platform Owner will raise a Swift Evolution proposal for
```

Comment:
What is a "Platform"? Is "Linux" a platform, or are "Debian", "Fedora", and "Arch Linux" separate platforms? At least between Debian and Arch, there is a considerable difference in their release strategies that impact the ABI stories, ranging from, "libraries should be held stable for one or two releases" to "rolling-release. ABI is whatever you happened to have downloaded at the moment. Partial-upgrades are not supportable".

Reply:
I think Linux probably _is_ a platform, but I appreciate it's a complicated case and will require some special handling.

Reply:
I'm 100% on-board with nailing down the calling conventions, but still have questions about the library ABI/binary compatibility story.

Will we be requiring full bi-directional binary compatibility like on Darwin where you can build against a new runtime and run it on older runtimes (assuming minimum deployment targets work out) as well as building against older runtimes and running it on newer runtimes?
Or are we okay with something more like glibc/libstdc++ where something built against the older runtime will continue working on newer runtimes, but not the other way around?

Reply:
Personally I think we should aim high here; if we can make Swift itself work the way it does on Darwin, that's actually quite a compelling feature for Linux development — Swift programs won't have the deployment problems that C/C++ programs do.

I realise that's not easy and will require `@availability` support and some kind of Swift minimum target version support from the compiler too.